### PR TITLE
Add print terminal to page depending on if printing is supported.

### PR DESCRIPTION
**Pull request by kchen05**
Screenshots for reference: 
image
image
image
image

### CODE REVIEW

**Comment by kchen05** [comment]
@eyurko @seanlip PTAL, switched to new branch as the other was getting somewhat cluttered.

File: client/data/domain/TipObjectFactory.js [hunk 1]
```diff
@@ -18,14 +18,23 @@
...
     /**
      * Constructor for Tip objects.
      *
      * @param {dict} tipDict
      * @constructor
      */
     var Tip = function(tipDict) {
+      /**
+       * The boolean regarding whether printing needs to be disabled in order
+       * for the tip to be triggered.
```
Comment (eyurko) [inline client/data/domain/TipObjectFactory.js]: nit: triggered -> activated / shown

File: client/data/domain/TipObjectFactory.js [hunk 2]
```diff
@@ -70,6 +88,12 @@ tieData.factory('TipObjectFactory', [
      * @returns {boolean}
      */
     Tip.prototype.isTriggeredBy = function(codeLines) {
```
Comment (eyurko) [inline client/data/domain/TipObjectFactory.js]: Also nit -> Activated

File: client/question/services/UnpromptedFeedbackManagerServiceSpec.js [hunk 3]
```diff
@@ -106,24 +111,55 @@ describe('UnpromptedFeedbackManagerService', function() {
...
+        LANGUAGE_PYTHON, 'import', taskId);
       expect(feedbackParagraphs.length).toBe(1);
       expect(feedbackParagraphs[0].isTextParagraph()).toBe(true);
-      expect(feedbackParagraphs[0].getContent()).toMatch(
-        'We noticed that you\'re using a print statement');
+      expect(feedbackParagraphs[0].getContent()).toBe(
+        'For this question, you do not need to import libraries.');
     });
+
+    it('should not trigger a warning for print statements if print is enabled',
```
Comment (eyurko) [inline client/question/services/UnpromptedFeedbackManagerServiceSpec.js]: trigger -> activate

File: client/data/domain/TipObjectFactory.js (hunk 1 above)
Comment (kchen05) [reply client/data/domain/TipObjectFactory.js]: Fixed

File: client/data/domain/TipObjectFactory.js (hunk 2 above)
Comment (kchen05) [reply client/data/domain/TipObjectFactory.js]: Fixed

File: client/question/services/UnpromptedFeedbackManagerServiceSpec.js (hunk 3 above)
Comment (kchen05) [reply client/question/services/UnpromptedFeedbackManagerServiceSpec.js]: Fixed

File: client/data/domain/TipObjectFactory.js [hunk 4]
```diff
@@ -18,14 +18,23 @@
...
+  function(PrintTerminalService) {
     /**
      * Constructor for Tip objects.
      *
      * @param {dict} tipDict
      * @constructor
      */
     var Tip = function(tipDict) {
+      /**
+       * The boolean regarding whether printing needs to be disabled in order
```
Comment (seanlip) [inline client/data/domain/TipObjectFactory.js]: Drop "The boolean regarding"

File: client/question/components/LearnerViewDirective.js [hunk 5]
```diff
@@ -280,7 +289,14 @@ tie.directive('learnerView', [function() {
         .tie-coding-terminal {
           display: flex;
           font-size: 13px;
-          height: 528px;
+          height: 100%;
+          position: relative;
+          width: 662px;
+        }
+        .print-mode .tie-coding-terminal{
```
Comment (seanlip) [inline client/question/components/LearnerViewDirective.js]: nit: add space before {

File: client/question/components/LearnerViewDirective.js [hunk 6]
```diff
@@ -300,6 +316,51 @@ tie.directive('learnerView', [function() {
           margin: 8px;
           white-space: normal;
         }
+        .tie-print-title {
+          padding-top: 3px;
```
Comment (seanlip) [inline client/question/components/LearnerViewDirective.js]: In general try to alphabetize CSS attrs within each rule (here and below).

File: client/question/components/LearnerViewDirective.js [hunk 7]
```diff
@@ -582,6 +645,12 @@ tie.directive('learnerView', [function() {
           {themeName: THEME_NAME_DARK}
         ];
 
+        /**
+         * Defines whether printing is supported, and thus whether the print
+         * terminal should be displayed.
+         */
+        $scope.printSupported = PrintTerminalService.isPrintingSupported();
```
Comment (seanlip) [inline client/question/components/LearnerViewDirective.js]: $scope.printingIsSupported

File: client/data/domain/TipObjectFactory.js [hunk 8]
```diff
@@ -64,12 +82,18 @@ tieData.factory('TipObjectFactory', [
...
 
     /**
-     * Whether the given lines of code trigger the tip.
+     * Whether the given lines of code activate the tip.
      *
      * @param {Array<string>} codeLines The lines of code to examine.
      * @returns {boolean}
      */
-    Tip.prototype.isTriggeredBy = function(codeLines) {
+    Tip.prototype.isActivatedBy = function(codeLines) {
```
Comment (seanlip) [inline client/data/domain/TipObjectFactory.js]: Probably should add Karma tests for this.

Comment (kchen05) [reply client/data/domain/TipObjectFactory.js]: Added Karma tests.

**Comment by seanlip** [comment]
LGTM. Thanks!

**Pull request merged**